client/my-sites/themes/index.node.js

```diff
@@ -4,16 +4,19 @@
 import config from 'config';
 import { makeLoggedOutLayout } from 'controller';
 import { details, fetchThemeDetailsData } from './controller';
-import { setSection } from 'state/ui/actions';
+import { dispatchSetSectionMiddlewareFactory } from 'lib/react-helpers';
 
-function dispatchSetSection( context, next ) {
-	context.store.dispatch( setSection( {
-		name: 'themes',
-		group: 'sites',
-		secondary: true,
-	} ) );
-	next();
-}
+const dispatchThemesWithSidebar = dispatchSetSectionMiddlewareFactory( {
```

Comment:
Ideally, we won't invoke this directly here, but call it automatically inside `server/isomorphic-routing` (as I said here: https://github.com/Automattic/wp-calypso/pull/3913#issuecomment-196337726) with data from `sections.js` much like #3913 does on the client side, so we don't duplicate functionality.